### DISCUSSION TITLE
Lucide Icons for Figma

### DISCUSSION THREAD

**Discussion by balazsbarta** [lucide-icons/lucide#1233] [Show and tell]
Hello, everyone!

I utilized this repo to create a Figma file. All of the icons are components that you can publish as a library and then use like any other component in your work.

I've also documented each icon with the official tags so that you can quickly find the desired icon. All icons have been flattened and united to avoid color override issues. If you want, you can also easily change the stroke weights.

I also keep the file up to date, as my time permits.

If you liked the library, please click the like button at the top of the community page. Thank you!

**You can duplicate it [here](https://www.figma.com/community/file/1204720733890022011)**

**Comment by karsa-mistmere** [comment]
Nice work, Balázs, component support is actually something we'd like to explore in the future directly from our official Figma plugin. 😊

**Reply by uixray** [reply]
The disadvantage of working with icons through the plugin is that you have to disassemble the outline for their scaling in components. I would suggest putting the pack in a separate file, where the vector is wrapped in a separate container. For design systems, this is the most convenient approach
https://www.figma.com/file/eFybvt4PD3Awp1VJsQ19GS/%F0%9F%92%AB-Icons-Pack-%E2%80%94-Lucide-v0.252.0?type=design&t=CTJELuhDVazsmDIg-6

**Reply by karsa-mistmere** [reply]
@uixray I get where you're coming from, but we are extremely unlikely to do this, since one of the primary selling points of Lucide is that the stroke width is customisable, outlining the strokes would completely obliterate this feature.

**Reply by uixray** [reply]
![image](https://github.com/lucide-icons/lucide/assets/55608829/593da667-a47a-4136-8c8f-05ecbd5916ee)
I would be happy to use the contour version, but it does not work well with the Union selection feature, which will keep the color of the icon when changing its vector, so maybe it is worth adding an additional version? Or is it possible to optionalize icons in such a way that vector points did not overlap?

**Reply by balazsbarta** [reply]
@karsa-mistmere @uixray It is not necessary to outline the stroke; simply flatten the icons. In that case, you can still change the stroke weight, but the icons are now made up of a single vector object. In my Lucide Icons file, I took the same approach.

<img width="1625" alt="CleanShot 2023-06-26 at 10 51 18@2x" src="https://github.com/lucide-icons/lucide/assets/8573183/a7754a40-ccb4-495a-844d-75da17def141">

**Reply by uixray** [reply]
But if you need to replace the vector with a newer version,in all duplicates the color will roll back to the original(

**Reply by balazsbarta** [reply]
@uixray No, it's working properly. Just get my file and give it a try. It keeps the color, stroke and size, because each icon is made up of only one "Vector" named object. I use this method for all of my design systems and haven't had any problems with it yet.

**Reply by uixray** [reply]
I used to do that, too, but when you change the vector in a parent, the offspring won’t keep the color if it’s changed
Try — https://www.figma.com/file/PhFe37VRaZ7vIlf4zN0VWJ/Example?type=design&node-id=0%3A1&mode=design&t=s1MWCGbAy88VEGd9-1

**Reply by balazsbarta** [reply]
@uixray Now I know what's going wrong. But there is also a way to fix this. Move the icons to a separate library (need pro subscription) and update them there. Then, in your design file, just add that library. It should work in this case.

**Reply by uixray** [reply]
I tested it and it really works!! Thank you)If I find anything else, I will let you know

**Reply by mnlfischer** [reply]
@karsa-mistmere I think it is possible to create an "build file" button that creates icon components in Figma programmatically. 
Maybe this could be a [good starting point](https://medium.com/uber-design/how-i-programmatically-built-256-new-design-system-components-in-figma-84ee26d119c1)

**Reply by karsa-mistmere** [reply]
@mnlfischer I've already implemented this feature, it's just going to take some more time to roll out our new and shiny Figma plugin. 🙂

**Reply by mnlfischer** [reply]
@karsa-mistmere appreciate your work! thanks!

**Reply by mnlfischer** [reply]
@karsa-mistmere if I can help with something, let me know!

**Comment by ygd** [comment]
@balazsbarta I am planning to make a plugin for Adobe XD. I think XD and Figma plugins are not so far away. Can you give me an idea of how much work it is? I want to have a rough idea how big the project is going to be.
